### PR TITLE
[GPU] Align permute axis format with IE

### PR DESCRIPTION
**Pull request by vladimir-paramuzov**
### Details:
 - Align permute params with ngraph for further shape inference reuse

### CODE REVIEW

**Comment by ababushk** [comment]
cldnn_unit stage was killed by Jenkins due to log size exceeded 1GB.
Endless stream of this:
```
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO: ../../repos/openvino/src/plugins/intel_gpu/tests/test_cases/depth_to_space_gpu_test.cpp:262: Failure
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO: Expected equality of these values:
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO:   output_ptr_ref[i]
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO:     Which is: 15104
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO:   output_ptr[i]
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO:     Which is: 14848
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO: ../../repos/openvino/src/plugins/intel_gpu/tests/test_cases/depth_to_space_gpu_test.cpp:262: Failure
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO: Expected equality of these values:
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO:   output_ptr_ref[i]
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO:     Which is: 45056
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO:   output_ptr[i]
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO:     Which is: 13312
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO: ../../repos/openvino/src/plugins/intel_gpu/tests/test_cases/depth_to_space_gpu_test.cpp:262: Failure
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO: Expected equality of these values:
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO:   output_ptr_ref[i]
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO:     Which is: 46592
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO:   output_ptr[i]
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO:     Which is: 14592
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO: ../../repos/openvino/src/plugins/intel_gpu/tests/test_cases/depth_to_space_gpu_test.cpp:262: Failure
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO: Expected equality of these values:
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO:   output_ptr_ref[i]
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO:     Which is: 47872
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO:   output_ptr[i]
[2022-03-31T15:31:35.228Z] [2022-03-31 15:31:34,951] [139645333100288] cldnn_unit_tests-0 INFO:     Which is: 48128
```

**Comment by vladimir-paramuzov** [comment]
@ababushk I see, thanks. One of the tests has huge input and uses EXPECT_EQ to check the output... Will fix it soon.

File: src/plugins/intel_gpu/src/graph/impls/ocl/permute.cpp [hunk 1]
```diff
@@ -15,6 +15,33 @@ using namespace cldnn;
 namespace cldnn {
 namespace ocl {
 
+namespace {
+// This helper function is needed to convert permute order from IE format (bfyx) into cldnn format (bfxy)
+inline std::vector<uint16_t> convert_permute_order(const std::vector<uint16_t>& ie_order, size_t rank = 0) {
+    std::vector<uint16_t> ie_order_aligned = ie_order;
+    // if order size is less than 4 - fill the rest with just copy
+    rank = std::max(rank, (size_t)4);
+    for (auto o = ie_order_aligned.size(); o < rank; o++)
```
Comment (lznamens) [inline src/plugins/intel_gpu/src/graph/impls/ocl/permute.cpp:24]: My point is that _order_idx_ instead of _o_ looks better.

**Review by lznamens — APPROVED** [review]
LGTM.

**Pull request merged**